Parser.cpp

```diff
@@ -11,6 +11,91 @@ namespace fs = std::filesystem;
 namespace commonItems
 {
 std::string getNextLexeme(std::istream& theStream);
+
+
+
+bool registeredRegex::match(const std::string& lexeme, std::istream& theStream) const {
+	if (!std::regex_match(lexeme, regex))
+		return false;
+	else {
```

Comment:
Drop the else, the check works as an early return:

```
   if (!std::regex_match(lexeme, regex))
      return false;

   function(lexeme, theStream);
   return true;
```

Ditto for the other functions